cli/azd/pkg/azdo/project.go

```diff
@@ -0,0 +1,190 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azdo
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/environment"
+	"github.com/azure/azure-dev/cli/azd/pkg/input"
+	"github.com/microsoft/azure-devops-go-api/azuredevops"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/core"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/operations"
+)
+
+// returns a process template (basic, agile etc) used in the new project creation flow
+func getProcessTemplateId(ctx context.Context, client core.Client) (string, error) {
+	processArgs := core.GetProcessesArgs{}
+	processes, err := client.GetProcesses(ctx, processArgs)
+	if err != nil {
+		return "", err
+	}
+	process := (*processes)[0]
+	return process.Id.String(), nil
+}
+
+// creates a new Azure Devops project
+func createProject(ctx context.Context, connection *azuredevops.Connection, name string, description string, console input.Console) (*core.TeamProjectReference, error) {
+	coreClient, err := core.NewClient(ctx, connection)
+	if err != nil {
+		return nil, err
+	}
+
+	processTemplateId, err := getProcessTemplateId(ctx, coreClient)
+	if err != nil {
+		return nil, fmt.Errorf("error fetching process template id %w", err)
+	}
+
+	capabilities := map[string]map[string]string{
+		"versioncontrol": {
+			"sourceControlType": "git",
+		},
+		"processTemplate": {
+			"templateTypeId": processTemplateId,
+		},
+	}
+	args := core.QueueCreateProjectArgs{
+		ProjectToCreate: &core.TeamProject{
+			Description:  &description,
+			Name:         &name,
+			Visibility:   &core.ProjectVisibilityValues.Private,
+			Capabilities: &capabilities,
+		},
+	}
+	res, err := coreClient.QueueCreateProject(ctx, args)
+	if err != nil {
+		return nil, err
+	}
+
+	operationsClient := operations.NewClient(ctx, connection)
+
+	getOperationsArgs := operations.GetOperationArgs{
+		OperationId: res.Id,
+	}
+
+	projectCreated := false
+	maxCheck := 10
+	count := 0
+
+	for !projectCreated {
+		operation, err := operationsClient.GetOperation(ctx, getOperationsArgs)
+		if err != nil {
+			return nil, err
+		}
+
+		if *operation.Status == "succeeded" {
+			projectCreated = true
+		}
+
+		if count >= maxCheck {
+			return nil, fmt.Errorf("error creating azure devops project %s", name)
+		}
+
+		count++
+		time.Sleep(800 * time.Millisecond)
+	}
+
+	project, err := getAzdoProjectByName(ctx, connection, name)
+	if err != nil {
+		return nil, err
+	}
+	return project, nil
+}
+
+// prompts the user for a new AzDo project name and creates the project
+// returns project name, project id, error
+func GetAzdoProjectFromNew(ctx context.Context, repoPath string, connection *azuredevops.Connection, env *environment.Environment, console input.Console) (string, string, error) {
```

Comment:
Have you considered groups these functions of this module up into a struct with instance methods to avoid having to pass all the references around for connection, env, console, etc.?